src/client/presenceColor.ts

```diff
@@ -0,0 +1,7 @@
+import ColorHash from 'color-hash';
+import { Presence } from '../types';
+
+export const assignUserColor = (presence: Presence, id: string): Presence => {
```

Comment:
The concept is almost correct - I would suggest to have this just return the color, rather than mutating the presence object.